presto-main/src/main/java/com/facebook/presto/server/ResourceGroupStateInfoResource.java

```diff
@@ -108,9 +112,9 @@ public void getResourceGroupInfos(
                         summarizeSubgroups,
                         includeStaticSubgroupsOnly)).build());
             }
-            catch (NoSuchElementException | IllegalArgumentException e) {
-                asyncResponse.resume(Response.status(NOT_FOUND).build());
-            }
+        }
+        catch (NoSuchElementException | IllegalArgumentException e) {
+            asyncResponse.resume(Response.status(NOT_FOUND).build());
```

Comment:
I think this piece of code could send a double `NOT_FOUND` response 🤔 

In the flow, it will enter the catch and give the `NOT_FOUND` and then it will send the `NOT_FOUND` response again because of the line 119

Reply:
I was wondering the last line of `asyncResponse.resume()`. after adding `return` for each `resume` the last is not needed.